nodeup/pkg/model/kubeapiserver.go

```diff
@@ -127,22 +141,30 @@ func (b *KubeAPIServerBuilder) writeAuthenticationConfig(c *fi.ModelBuilderConte
 			Type:     nodetasks.FileType_File,
 		}
 		c.AddTask(t)
+
 		return nil
-	} else {
-		return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
 	}
+
+	return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
```

Comment:
BTW the main reason I like not having this as a fall-through is that then it is a compilation error if you forget to `return nil` in one of the branches, if each of the branches is expected to exit.